src/turtle_setter.cpp

```diff
@@ -0,0 +1,187 @@
+#include "turtlesim/Pose.h"
+#include "software_training_assignment/resetTurtle.h"
+#include "software_training_assignment/distance.h"
+#include "software_training_assignment/SeekTurtleAction.h"
+#include "geometry_msgs/Twist.h"
+#include "actionlib/server/simple_action_server.h"
+#include <cmath>
+
+//This publisher is to be initalized with gemoetery_msgs/Twist !!
+ros::Publisher SeekTurtlePub; //Global since Main needs access to this
+
+constexpr double RESET_X = 25.0;
+constexpr double RESET_Y = 10.0;
+
+
+struct Waypoint
+{
+    float x = 0;
+    float y = 0;
+    float distance = 0;
+    float angle = 0;
+};
+
+static Waypoint movingTurtleWP; //static to keep it file-scope
+
+float distance(float x, float y)
+{
+    return pow(pow(x,2) + pow(y,2), 1/float(2));
+}
+
+float arctanAbsolute(float x, float y)
+{
+    float DEGREES_IN_QUADRANT = 90;
+    float arctangent = atan(x/y)*M_PI;
+    if(y > 0 && x < 0)
+    {
+        arctangent = -arctangent+DEGREES_IN_QUADRANT*1; //Quadrant 2
+    }
+    else if(y < 0 && x < 0)
+    {
+       arctangent = arctangent+DEGREES_IN_QUADRANT*2; //Quadrant 3
+    }
+    else if(y < 0 && x > 0)
+    {
+        arctangent = -arctangent+DEGREES_IN_QUADRANT*3; //Quadrant 4
+    }
+
+    return arctangent;
+}
+
+void calculateDistance(Waypoint targetWaypoint, Waypoint currentWaypoint, Waypoint &resultWaypoint)
+{
+    resultWaypoint.x = targetWaypoint.x - currentWaypoint.x;
+    resultWaypoint.y = targetWaypoint.y - currentWaypoint.y;
+    resultWaypoint.distance = distance(resultWaypoint.x, resultWaypoint.y);
+    resultWaypoint.angle = arctanAbsolute(resultWaypoint.x, resultWaypoint.y);
+}
+
+void updateDistanceMsg(ros::NodeHandle &n, Waypoint DistanceStruct)
+{
+    ros::Publisher publisher = n.advertise<software_training_assignment::distance>("Distances",1000);
+    software_training_assignment::distance msg;
+    msg.x = DistanceStruct.x;
+    msg.y = DistanceStruct.y;
+    msg.dist = DistanceStruct.distance;
+    publisher.publish(msg);
+}
+
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS******/
+bool clearTurtle(std::string name) //clears the existing turtle
+{
+    turtlesim::Kill killService; //Service to kill the turtle
+    killService.request.name = name;
+    return ros::service::call("kill", killService); //Makes sure the turtle dies before exiting the loop
+
+}
+
+
+bool spawnTurtle(std::string name, double x, double y, double angle)
+{
+    turtlesim::Spawn spawnService;
+    spawnService.request.name = name;
+    spawnService.request.x = x;
+    spawnService.request.y = y;
+    spawnService.request.theta = angle;
+    return ros::service::call("spawn", spawnService);
+}
+
+/******CLEAR & SPAWN TURTLE FUNCTIONS END******/
+
+bool resetTurtlePosition(software_training_assignment::resetTurtle::Request &request, software_training_assignment::resetTurtle::Response &response)
+{
+    
+    std::string name = request.name;  //QUESTION: Why do I not need to dereference this??
```

Comment:
this is not a pointer, its passed by reference and therefore you can access the members directly

Reply:
Ah I see, thanks!!